### PR TITLE
12.0 mig account journal lock date

### PR DESCRIPTION
**Pull request by blaggacao**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Congratulations on your first module migration.

Check CIs status. Maybe dependencies are not yet migrated?

You can squash together adjacent "OCA Transbot.." commits using https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests

And finally, a new method for generating the README easily out of fragments is available with https://github.com/OCA/maintainer-tools/#readme-generator

**Comment by blaggacao** [comment]
:+1: Is this an automated message ? :smile:

**Comment by blaggacao** [comment]
>You can squash together adjacent "OCA Transbot.." commits using https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests

Kind of a case for an `oca-squash-translation-commits` helper? :wink:

**Comment by pedrobaeza** [comment]
No, I do it manually. Some people suspects that I have minions working for me, but I can assure it's me, hahaha. It's just my "extreme GTD" technique that annoys you sometimes :wink:

**Comment by blaggacao** [comment]
@pedrobaeza Every time less so :wink: I'd like to pre-validate something of importance (I feel) about OCA with you, is there a back channel where I can reach out to you?

**Comment by pedrobaeza** [comment]
> Kind of a case for an oca-squash-translation-commits helper? wink

Yeah, it can be, but this is not going to happen anymore with Weblate. The idea is to remove from each version unneeded iteration commits. That can happen with Transifex, or incremental translations on Weblate, or even old Launchpad system, or several commit from same author that are now meaningless from being split, but of course always preserving authors. With this, on 13.0 we will have very clean branches that don't take a lot to be pulled and the rest of git operations.

**Comment by pedrobaeza** [comment]
> I'd like to pre-validate something of importance (I feel) about OCA with you, is there a back channel where I can reach you?

Asynchronously you can contact by mail in my corporate mail, and I'm also in hangouts with that one for synchronous chats, but that's more difficult due to timeslots.

File: account_journal_lock_date/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    'name': 'Account Journal Lock Date',
+    'summary': """
+        Lock each journal independently""",
+    'version': '12.0.1.0.0',
+    'license': 'AGPL-3',
+    'author': 'ACSONE SA/NV,Odoo Community Association (OCA)',
+    'website': 'https://acsone.eu/',
```
Comment (nikul-serpentcs) [inline account_journal_lock_date/__manifest__.py:12]: @blaggacao Should be `https://github.com/OCA/account-financial-tools`

File: account_journal_lock_date/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    'depends': ['account'],
+    'data': [
+        'views/account_journal.xml',
+    ],
+    'demo': [
```
Comment (nikul-serpentcs) [inline account_journal_lock_date/__manifest__.py:17]: @blaggacao If nor any file then remove it

File: account_journal_lock_date/views/account_journal.xml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2017 ACSONE SA/NV
+     Copyright 2018 XOE Corp. SAS <https://xoe.solutions>
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+
+<odoo>
+<data>
```
Comment (nikul-serpentcs) [inline account_journal_lock_date/views/account_journal.xml:7]: @blaggacao remove `<data>`

File: account_journal_lock_date/models/account_journal.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+from odoo.tools.misc import DEFAULT_SERVER_DATE_FORMAT
+from odoo.exceptions import ValidationError
+
+
+class AccountJournal(models.Model):
+
+    _inherit = 'account.journal'
+
+    journal_lock_date_period_offset = fields.Integer(
+        string="Lock date offset", default=0,
```
Comment (nikul-serpentcs) [inline account_journal_lock_date/models/account_journal.py:17]: @blaggacao no need to `default = 0`

https://odoo-new-api-guide-line.readthedocs.io/en/latest/fields.html#integer

File: account_journal_lock_date/models/account_journal.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        help="Set a stricter (later) lock date for this journal with this "
+             "offest in days respective of the period lock date."
+    )
+    has_period_lock_date = fields.Boolean(compute='_compute_lock_date')
+    journal_lock_date = fields.Date(
+        string="Lock date", readonly=True, store=True,
```
Comment (nikul-serpentcs) [inline account_journal_lock_date/models/account_journal.py:23]: @blaggacao no need to `readonly=True`, Compute always ` readonly`

**Review by nikul-serpentcs — CHANGES_REQUESTED** [review]
@blaggacao Improve Code

**Comment by pedrobaeza** [comment]
As @blaggacao has moved to other software AFAIK, and there's #831 from active contributor, we are moving to use that one if you don't mind. If you don't agree, just tell me, as you have priority being the first PR. This module had a problem in consistency with the name, as it should be independent from permanent lock.

**Pull request closed without merging**